### PR TITLE
Signing providers

### PR DESCRIPTION
**Pull request by andreibancioiu**
#### Description of the pull request (what is new / what has changed)

Document usage of erdjs' signing providers (for dApps).

#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [x] document new feature
- [ ] update documentation that is not relevant anymore
- [x] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+Before performing any operation, make sure to initialize the provider (also, the Elrond application has to be open on the device):
+
+```
+await provider.init();
+```
+
+### Login
+
+Before asking the user to login using the Ledger, you may want to display all the available addresses on the device, and let the user choose one of them:
```
Comment (bogdan-rosianu) [inline docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:306]: `log in`*

Comment (bogdan-rosianu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:306]: also, all the addresses aren't displayed on the device, but rather on the website/application. The device only displays one of them at a time

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+```
+import qs from "qs";
+
+const queryString = window.location.search.slice(1);
+const params = qs.parse(queryString);
+console.log(params.address);
+```
+
+In order to logout, do as follows:
```
Comment (bogdan-rosianu) [inline docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:52]: `log out`*

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+Then, ask the user to login:
+
+```
+const address = await provider.login();
+
+console.log(address);
+console.log(provider.account);
+```
+
+In order to logout, do as follows:
```
Comment (bogdan-rosianu) [inline docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:152]: `log out`*

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+});
+
+await provider.signMessage(message);
+
+console.log(message.toJSON());
+```
+
+## Verifying the signature of a login token
+
+As previously mentioned, a dApp (and its backend) might want to reliably assign an off-chain user identity to an Elrond address. On this purpose, the signing providers allow an _login token_ to be used within the login flow - this token is signed using the wallet of the user. Afterwards, a backend application would normally verify the signature of the token, as follows:
```
Comment (bogdan-rosianu) [inline docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:380]: `a` _login token_

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+    const ok = verifier.verify(message);
+    if (ok) {
+        return `The bearer of the token [${authToken}] is also the owner of the address [${address}].`;
+    }
+
+    return "Verification failed.";
+}
+```
+
+:::important
```
Comment (bogdan-rosianu) [inline docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:407]: could have been a note

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+
+This page will guide you through the process of integrating the **erdjs signing providers** in a dApp which isn't based on `dapp-core`.
+
+:::important
+Note that for most purposes, **we recommend using [dapp-core](https://github.com/ElrondNetwork/dapp-core)** instead of integrating the signing providers on your own.
+:::
+
+:::important
+The code samples depicted on this page can also be found on the [**erdjs examples repository**](https://github.com/ElrondNetwork/elrond-sdk-erdjs-examples).
+:::
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:14]: Added.

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+
+```
+const connectorUri = await provider.login();
+QRCodeModal.open(connectorUri);
+```
+
+Once the user confirms the login, the `onClientLogin()` callback (declared above) is executed.
+
+
+In order to logout, do as follows:
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:251]: Fixed.

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md (hunk 1 above)
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:306]: Rephrased.

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md (hunk 2 above)
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:52]: Fixed.

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md (hunk 3 above)
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:152]: Fixed.

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md (hunk 4 above)
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:380]: Fixed.

File: docs/sdk-and-tools/erdjs/erdjs-signing-providers.md (hunk 5 above)
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-signing-providers.md:407]: Note it is :+1:

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**